asyncsleepiq/fuzion/foundation.py

```diff
@@ -91,19 +91,19 @@ async def init_presets(self, data: dict[str, Any]) -> None:
             options.append(PRESET_READ)
 
         if self.type in ["single", "easternKing"]:
-            self.presets = [SleepIQPreset(self._api, self.bed_id, Side.NONE)]
+            self.presets = [SleepIQFuzionPreset(self._api, self.bed_id, Side.NONE, options)]
         else:
             self.presets = [
-                SleepIQPreset(self._api, self.bed_id, Side.LEFT),
-                SleepIQPreset(self._api, self.bed_id, Side.RIGHT),
+                SleepIQFuzionPreset(self._api, self.bed_id, Side.LEFT, options),
+                SleepIQFuzionPreset(self._api, self.bed_id, Side.RIGHT, options),
             ]
 
-        await self.update_presets([])
+        await self.update_presets({})
 
     async def fetch_features(self) -> None:
         """Update list of features available for foundation from API."""
-        vals = await self._api.bamkey(self.id, "GetSystemConfiguration")
-        for k, v in zip(FEATURE_NAMES, vals.split()):
+        vals = await self._api.bamkey(self.bed_id, "GetSystemConfiguration")
+        for k, v in zip(FEATURE_NAMES, vals.split()[1:]): # Slice off the first item (often "dual" - not a feature we track)

```

Comment:
I am not sure this is the best way to solve this - but the very first item in the response for `GetSystemConfiguration` is `dual`, and that doesn't actually map to any of the features we have listed above, so I just opted to slice it off and map the remaining items

Can probably be made more clear, or we could opt to add a feature for this!

(FWIW, it looks like this)
```
{"cdcResponse":"PASS:dual yes yes yes yes heat_cool yes yes yes yes yes yes yes yes yes yes"}
```

@kbickar Maybe we should just strip `PASS:dual` instead of just `pass` when we use the `bamkey` function?

Reply:
We might need the "dual" at some point - could add something to the start of the FEATURE_NAMES for the type

Reply:
Yeah, that works too - I wasn't quite sure what to call the feature, but I can add something in just so we're capturing the value in the feature array

Reply:
36d1b42